app_history.d.ts

```diff
@@ -0,0 +1,99 @@
+interface Window {
+  readonly appHistory: AppHistory;
+}
+
+interface AppHistory extends EventTarget {
+  readonly current: AppHistoryEntry|null;
+  readonly transition: AppHistoryTransition|null;
+
+  entries(): AppHistoryEntry[];
+
+  readonly canGoBack: boolean;
+  readonly canGoForward: boolean;
+
+  navigate(url: string, options?: AppHistoryNavigateOptions): Promise<void>;
```

Comment:
I believe you need the overload for this for both options, right? 

```.ts
 navigate(options: AppHistoryFullOptions): Promise<void>;
 navigate(
    url?: string,
    options?: AppHistoryNavigateOptions
  ): Promise<void>;
  
 interface AppHistoryPushOrUpdateFullOptions extends AppHistoryNavigateOptions {
   url?: string;
 }
  ```

Reply:
Oh also, I believe that you can call `navigate()` with no options as well, correct?

Reply:
The old no-url navigate() variant still exists in Chrome, but not in the spec; in the spec it was replaced by reload().